File: cpp/pangram/pangram.cpp
#include "pangram.h"
#include <bitset>
#include <locale>
#include <string>

bool pangram::is_pangram(const std::string &str) {
    std::bitset<26> letters;

    for (auto c : str) {
        if (std::isalpha(c, std::locale::classic())) {
            const auto pos = std::tolower(c, std::locale::classic()) - 'a';
            letters.set(pos);
        }
    }

    return letters.all();
}

File: go/pangram/pangram.go
package pangram

import "strings"

func IsPangram(s string) bool {
	s = strings.ToLower(s)
	count := map[rune]bool{}

	for _, c := range s {
		if c >= 'a' && c <= 'z' {
			count[c] = true
		}
	}

	return len(count) == 26
}
